x/executionlayer/types/types.go

```diff
@@ -40,6 +45,6 @@ func NewPublicKey(base64PublicKey string) (*PublicKey, error) {
 // ToPublicKey convert sdk.AccAddress to PublicKey appending null padding.
 // we currently use sdk.AccAddress as public key for PoC.
 // This should be removed later.
-func ToPublicKey(accAddr sdk.AccAddress) PublicKey {
-	return append(accAddr.Bytes(), make([]byte, 12)...)
+func ToPublicKey(accAddr []byte) PublicKey {
```

Comment:
Is there a special reason to break type safety?

Reply:
To use Val.Address
Is there a better way to use multiple types?

Reply:
One of the rudimentary reason is Cosmos SDK separates validator address and account address while CasperLabs does not.

In my opinion, someone might think the way below is redundant, how about like
```go
func ToPublicKey(accAddr sdk.AccAddress) PublicKey {
	return append(accAddr.Bytes(), make([]byte, 12)...)
func ToPublicKey(valAddr sdk.ValAddress) PublicKey {
	return append(valAddr.Bytes(), make([]byte, 12)...)
```
for clear intent?

Reply:
How about this one? 
`
func ToPublicKey(addr sdk.Address) PublicKey {
	return append(addr.Bytes(), make([]byte, 12)...)
}
`

Reply:
https://github.com/hdac-io/friday/pull/47/commits/8b4d87d1a76679e8e91ebfc37db9be3482c243cd